.travis.yml

```diff
@@ -1,31 +1,98 @@
-language: python
+os: linux
 dist: bionic
-python:
-  - "2.7"
-  - "3.7"
+language: c++
+cache: ccache
 
-before_install:
-  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq g++-9
-  - export CXX="g++-9" CC="gcc-9"
-  - sudo apt-get install git
-  - sudo apt-get install ninja-build python3-pip python3-setuptools
+addons:
+  apt:
+    update: true
+    sources:
+      - sourceline: "ppa:ubuntu-toolchain-r/test"
+      - sourceline: "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic main"
+        key_url: "https://apt.llvm.org/llvm-snapshot.gpg.key"
+    packages:
+      - python3-pip
+      - ninja-build
+      - python3-setuptools
+      - ccache
+      - yasm
+  homebrew:
+    packages:
+      - ninja
+      - ccache
+      - yasm
+
+before_install: "sudo chown -R travis: $HOME/.ccache"
 
 install:
-  - pip install tox-travis
+  - ccache -s
   - pip3 install meson
+  - |
+    if type apt-get; then
+      case "$CC" in
+      gcc-7) sudo apt-get install -qq gcc-7 g++-7 ;;
+      gcc-8) sudo apt-get install -qq gcc-8 g++-8 ;;
+      gcc-9) sudo apt-get install -qq gcc-9 g++-9 ;;
+      clang-6.0) sudo apt-get install -qq clang-6.0 ;;
+      clang-10) sudo apt-get install -qq clang-10 ;;
+      esac
+    fi
+  - export CC="ccache $CC" CXX="ccache $CXX"
+
+after_script:
+  - ccache -s
 
-script:
-  - >
-    cd third_party/libsvm &&
-    make lib &&
-    cd -
-  - >
-    cd libvmaf &&
-    mkdir build && cd build &&
-    meson .. --buildtype release &&
-    ninja -vC . test &&
-    DESTDIR=$(pwd)/install ninja -vC . install &&
-    cd ../..
-  - tox -c python/ -- -v -p no:warnings -m 'main or lib' --doctest-modules
+matrix:
+  fast_finish: true
+  include:
+    - name: libvmaf GCC-7
+      env: CC=gcc-7 CXX=g++-7
+      script: &base-gcc-script
+        - mkdir libvmaf/build && cd libvmaf/build
+        - meson .. --buildtype release
+        - sudo ninja -v install
+        - sudo ninja -v test
+        - cd $TRAVIS_BUILD_DIR
+    - name: libvmaf GCC-9
+      env: CC=gcc-9 CXX=g++-9
+      script: *base-gcc-script
+    - name: libvmaf Clang-10
+      env: CC=clang-10 CXX=clang++-10
+      script: *base-gcc-script
+    - name: libvmaf AppleClang
+      os: osx
+      osx_image: xcode11.2
+      compiler: clang
+      script: *base-gcc-script
+    - name: Python3 Tox
+      language: python
+      python: "3.7"
+      env: CC=gcc-7 CXX=g++-7
+      cache: &python-cache
+        pip: true
+        directories:
+          - $HOME/.ccache
+      before_script: *base-gcc-script
+      script: &python-tox
+        - pip3 install tox-travis
+        - make -C third_party/libsvm -j lib
+        - tox -c python/ -- -v -p no:warnings -m 'main or lib' --doctest-modules
+    - name: libvmaf FFmpeg
```

Comment:
This looks good for now. I may later move this into a python test, to be aligned with the rest of the unittests.

Reply:
Sure, that would be good if possible, I'm not sure how "stable" my "test" is with just comparing the vmaf output to a previous generated number